pkg/endpoint/bpf.go

```diff
@@ -253,8 +253,13 @@ func (e *Endpoint) addNewRedirectsFromDesiredPolicy(ingress bool, desiredRedirec
 				var err error
 				redirectPort, err, finalizeFunc, revertFunc = e.proxy.CreateOrUpdateRedirect(e.aliveCtx, l4, proxyID, e, proxyWaitGroup)
 				if err != nil {
-					revertStack.Revert() // Ignore errors while reverting. This is best-effort.
-					return err, nil, nil
+					// Skip redirects that can not be created or updated.  This
+					// can happen when a listener is missing, for example.
+					// Policy will need to be regenerated when listeners are
+					// added or removed to reflect the change in listeners in
+					// the realized policy!
+					e.getLogger().WithError(err).Debug("Redirect rule skipped")
+					continue
```

Comment:
Previously this was an error path, but now it's no longer returning errors here. Could you elaborate on the change of behavior here and what the consequences are? Is debug level enough for logging?

Reply:
This is expected to happen when policy refers to a listener that does not exist yet, for example when restarting and k8s delivers the CNP before the related CEC. Related to this policy regeneration is now triggered when listeners are added/changed/deleted.

Debug level is enough to see that this happens as intended.

Reply:
I think 

> for example when restarting and k8s delivers the CNP before the related CEC

is good to add to the code comment, because otherwise it's not clear why.

Reply:
Reworded as:
```
					// Skip redirects that can not be created or updated.  This
					// can happen when a listener is missing, for example when
					// restarting and k8s delivers the CNP before the related
					// CEC.
					// Policy is regenerated when listeners are added or removed
					// to fix this condition when the listener is available.
					e.getLogger().WithField(logfields.L4PolicyID, proxyID).WithError(err).Debug("Redirect rule with missing listener skipped")
					continue
```